### DISCUSSION TITLE
RV32 Dhrystone benchmark - Error: Syscall 403 out of range

### DISCUSSION THREAD

**Discussion by satishkumar2003** [gem5/gem5#1333] [Q&A]
Hello! My query consists of two parts. The first one being "How do I build gem5.opt for a RISC-V 32 simulation". The build I have done now is following a stack overflow solution wherein I made the following replacements:

```python
RiscvISA.py  95: riscv_type = Param.RiscvType("RV32","RV32 or RV64") # RV64 --> RV32
RiscvISA.py 103: elen = Param.RiscvVectorElementLength(32, ....... ) # 64 --> 32
```

Following this replacement and build ( correct me if the build process was wrong ), I had cross-compiled a dhrystone benchmark (src: [Fleker's Dhrystone benchmark](https://github.com/Fleker/memristor-research/blob/master/experiments/dhrystone.c)) using riscv32-unknown-elf-gcc 13.1.0 as follows:

`
riscv32-unknown-elf-gcc -march=rv32gc -mabi=ilp32d -static -o dhrystone_rv32 dhrystone.c -lm -g
`

The **simple-riscv.py** config file in the learning_gem/part1 directory was updated with the new binary's location and was simulated. The simulation abruptly stops with a "**fatal: Syscall 403 out of range**" error
![Screenshot from 2024-07-06 11-14-14](https://github.com/gem5/gem5/assets/96340698/23e4a5b8-4a89-424a-bdb2-f8a9e3975ece)

An observation I made was that the config file was setting up an SEWorkload which led me to look into the se_workload.cc file which had a SyscallDescTable for SEWorkload::SyscallABI32 containing many ID's and their respective command. The ID 403 was not present in this Table as well:

![Screenshot from 2024-07-06 11-19-30](https://github.com/gem5/gem5/assets/96340698/cd64799b-8727-4e17-abeb-55846d68cffe)

My objective is to run an RV32 simulation for the dhrystone benchmark. Please correct me on the build method if its incorrect and also information on the syscall error. Any help on this would be very helpful! Thank you in advance!

**Comment by wslqk** [comment]
Hello, have you resolved this issue? I am facing the same problem while using riscv32-unknown-linux-gnu-gcc, encountering the error: src/sim/syscall_desc.hh:209: fatal: syscall 291 out of range.

**Comment by Juan-Gg** [comment]
I have the same issue 
(`src/sim/syscall_desc.hh:209: fatal: Syscall 291 out of range`) with gem5 version 24.1.0.1 and riscv32-unknown-linux-gnu-gcc (g04696df09) 14.2.0.
